pandas/io/common.py

```diff
@@ -448,9 +448,11 @@ def infer_compression(
     if compression in _compression_to_extension:
         return compression
 
-    msg = f"Unrecognized compression type: {compression}"
-    valid = ["infer", None] + sorted(_compression_to_extension)
-    msg += f"\nValid compression types are {valid}"
+    infer = {"infer": None}
+    msg = (
+        f"Unrecognized compression type: {compression}\n"
+        f"Valid compression types are {_compression_to_extension} and {infer}"
```

Comment:
This is changing the error message? Do we have a test that checks the message?

we have a few cases where using list literals produces this mypy error. This one is probably OK to ignore.

Reply:
No, we don't have a test for the message.